File: packages/ruby-sdk/lib/strait/operations/analytics.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for analytics and performance metrics.
    class AnalyticsService < BaseService
      # Get performance analytics.
      # GET /v1/analytics/performance
      def get_performance(query: nil)
        _request(:get, "/v1/analytics/performance", query: query)
      end
    end
  end
end

File: packages/python-sdk/strait/operations/analytics.py
"""Performance analytics operations."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class AnalyticsService(BaseService):
    def get_performance(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return self._request("GET", "/v1/analytics/performance", query=query)


class AsyncAnalyticsService(AsyncBaseService):
    async def get_performance(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return await self._request("GET", "/v1/analytics/performance", query=query)
